src/rma/destruction/views.py

```diff
@@ -99,3 +119,98 @@ def get_queryset(self):
         return prefiltered_qs.annotate(
             review_status=models.Subquery(review_status[:1])
         ).order_by("-created")
+
+
+class ReviewItemInline(InlineFormSetFactory):
+    model = DestructionListItemReview
+    fields = ["destruction_list_item", "text", "suggestion"]
+    formset_class = ReviewItemBaseFormset
+
+
+class ReviewCreateView(RoleRequiredMixin, CreateWithInlinesView):
+    model = DestructionListReview
+    form_class = ReviewForm
+    inlines = [ReviewItemInline]
+    template_name = "destruction/review_create.html"
+    success_url = reverse_lazy("destruction:reviewer-list")
+    role_permission = "can_review_destruction"
+
+    def get_destruction_list(self):
+        list_id = self.kwargs.get("destruction_list")
+        queryset = DestructionList.objects.filter(id=list_id)
+
+        try:
+            destruction_list = queryset.get()
+        except DestructionList.DoesNotExist:
+            raise Http404(_("No destruction list found matching the query"))
+
+        return destruction_list
+
+    def get_initial(self):
+        destruction_list = self.get_destruction_list()
+        initial = {
+            "author": self.request.user,
+            "destruction_list": destruction_list,
+        }
+
+        return initial
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        formset = context["inlines"][0]
+
+        destruction_list = self.get_destruction_list()
+        context.update(
+            {
+                "destruction_list": destruction_list,
+                "destruction_list_json": {
+                    "name": destruction_list.name,
+                    "author": destruction_list.author.get_full_name(),
+                    "created": timesince(destruction_list.created),
+                },
+                "formset_config": {
+                    "prefix": formset.prefix,
+                    **{
+                        field.name: int(field.value())
+                        for field in formset.management_form
+                    },
+                },
+            }
+        )
+        return context
+
+    def form_valid(self, form):
+        if "approve" in self.request.POST:
+            form.instance.status = ReviewStatus.approved
+
+        return super().form_valid(form)
+
+    def forms_valid(self, form, inlines):
+        response = super().forms_valid(form, inlines)
```

Comment:
```suggestion
    @transaction.atomic
    def forms_valid(self, form, inlines):
        response = super().forms_valid(form, inlines)
```

Important to wrap all this in a transaction, and only call the celery task in `transaction.on_commit`

Reply:
done